eclair-core/src/test/scala/fr/acinq/eclair/router/GraphSpec.scala

```diff
@@ -348,4 +348,28 @@ class GraphSpec extends AnyFunSuite {
     assert(paths(1).path == Seq(edgeCE, edgeEG, edgeGH))
     assert(paths(2).path == Seq(edgeCD, edgeDF, edgeFH))
   }
+
+  test("RoutingHeuristics.normalize when value in range") {
+    assert(Graph.RoutingHeuristics.normalize(value = 10, min = 0, max = 100) == 0.1)
+    assert(Graph.RoutingHeuristics.normalize(value = 20, min = 10, max = 200) == 0.05263157894736842)
+    assert(Graph.RoutingHeuristics.normalize(value = -11, min = -100, max = -10) == 0.9888888888888889)
+    assert(Graph.RoutingHeuristics.normalize(value = 0, min = 0, max = 100) == 1.0E-5)
+    assert(Graph.RoutingHeuristics.normalize(value = 10, min = 10, max = 200) == 1.0E-5)
+    assert(Graph.RoutingHeuristics.normalize(value = -100, min = -100, max = -10) == 1.0E-5)
+    assert(Graph.RoutingHeuristics.normalize(value = 9.1, min = 10, max = 200) == 1.0E-5)
+    assert(Graph.RoutingHeuristics.normalize(value = 100, min = 0, max = 100) == 0.99999)
+    assert(Graph.RoutingHeuristics.normalize(value = 200, min = 10, max = 200) == 0.99999)
+    assert(Graph.RoutingHeuristics.normalize(value = 105.2, min = 0, max = 100) == 0.99999)
+    assert(Graph.RoutingHeuristics.normalize(value = 0.001, min = 0, max = 100) == 1.0E-5)
+    assert(Graph.RoutingHeuristics.normalize(value = 10.00001, min = 10, max = 200) == 1.0E-5)
+    assert(Graph.RoutingHeuristics.normalize(value = 99.999999, min = 0, max = 100) == 0.99999)
+    assert(Graph.RoutingHeuristics.normalize(value = 199.999999934, min = 10, max = 200) == 0.99999)
+  }
+
+
+  test("Graph.RoutingHeuristics.normalize should throw exception if min > max") {
+    assertThrows[IllegalArgumentException](
+      Graph.RoutingHeuristics.normalize(value = 9, min = 10, max = 1)
+    )
+  }
```

Comment:
```suggestion
  test("RoutingHeuristics.normalize") {
    // value inside the range
    assert(Graph.RoutingHeuristics.normalize(value = 10, min = 0, max = 100) === (10.0 / 100.0) +- 0.001)
    assert(Graph.RoutingHeuristics.normalize(value = 20, min = 10, max = 200) === (10.0 / 190.0) +- 0.001)
    assert(Graph.RoutingHeuristics.normalize(value = -11, min = -100, max = -10) == (89.0 / 90.0) +- 0.001)
    
    // value on the bounds
    assert(Graph.RoutingHeuristics.normalize(value = 0, min = 0, max = 100) > 0)
    assert(Graph.RoutingHeuristics.normalize(value = 10, min = 10, max = 200) > 0)
    assert(Graph.RoutingHeuristics.normalize(value = -100, min = -100, max = -10) > 0)
    assert(Graph.RoutingHeuristics.normalize(value = 9.1, min = 10, max = 200) > 0)
    assert(Graph.RoutingHeuristics.normalize(value = 100, min = 0, max = 100) < 1)
    assert(Graph.RoutingHeuristics.normalize(value = 200, min = 10, max = 200) < 1)
    
    // value outside the range
    assert(Graph.RoutingHeuristics.normalize(value = 105.2, min = 0, max = 100) < 1)
    
    // Should throw exception if min > max
    assertThrows[IllegalArgumentException](
      Graph.RoutingHeuristics.normalize(value = 9, min = 10, max = 1)
    )
  }
```

Tests should not contain unexplainable constants, to get these constants you just ran the function so it just tests that the function returns what the function returned.
The fact that we clamp at 0.00001 is an implementation detail, not something we want to test. We only want to be in (0, 1).